griptape/tasks/text_query_task.py

```diff
@@ -11,6 +11,9 @@ class TextQueryTask(BaseTextInputTask):
     query_engine: BaseQueryEngine = field(kw_only=True)
     loader: TextLoader = field(default=Factory(lambda: TextLoader()), kw_only=True)
     namespace: Optional[str] = field(default=None, kw_only=True)
+    top_n: Optional[int] = field(default=None, kw_only=True)
 
     def run(self) -> TextArtifact:
-        return self.query_engine.query(self.input.to_text(), namespace=self.namespace, rulesets=self.all_rulesets)
+        return self.query_engine.query(
+            self.input.to_text(), namespace=self.namespace, rulesets=self.all_rulesets, top_n=self.top_n
```

Comment:
I believe this is failing static analysis because `query_engine` is a `BaseQueryEngine` that does not support `top_n` in its `query()` method. I think you'll need to update the base class to match.